### PR TITLE
Format during Return in a Brace Completion Session

### LINKED ISSUE #3447: Enter within {}s doesn't insert newline if } isn't set to format
1. Under Tools > Options, C#, Formatting, General, turn off "automatically format on }"
2. Type if (true) \<enter\> {
3. Observe the } was inserted after your caret
4. Press \<enter\>

*Expected:* you are in a blank line between two {]s
*Actual:*
capture

(filed on behalf of @Pilchie)

### PR DESCRIPTION
**Pull request by basoundr**
Fix #3447 Format should happen when Return is issued after starting a
Brace Completion session. We should not listen to
FeatureOnOffOptions.AutoFormattingOnCloseBrace when 'Return' is pressed
during the Brace Completion session

### CODE REVIEW

**Comment by basoundr** [comment]
Reviewers @Pilchie @jasonmalinowski @DustinCampbell @dpoeschl @rchande @balajikris @brettfo @davkean @jmarolf

**Comment by basoundr** [comment]
@dotnet-bot please retest this

**Comment by basoundr** [comment]
@dotnet-bot Retest this please

**Comment by basoundr** [comment]
@dotnet-bot retest this please

File: src/EditorFeatures/CSharp/AutomaticCompletion/Sessions/CurlyBraceCompletionSession.cs [hunk 1]
```diff
@@ -39,7 +39,7 @@ public CurlyBraceCompletionSession(ISyntaxFactsService syntaxFactsService, ISmar
 
         public override void AfterStart(IBraceCompletionSession session, CancellationToken cancellationToken)
         {
-            FormatTrackingSpan(session);
+            FormatTrackingSpan(session, true);
```
Comment (Pilchie) [inline src/EditorFeatures/CSharp/AutomaticCompletion/Sessions/CurlyBraceCompletionSession.cs]: Please add the named parameter so we can see what this literal means?

File: src/EditorFeatures/CSharp/AutomaticCompletion/Sessions/CurlyBraceCompletionSession.cs [hunk 2]
```diff
@@ -65,12 +65,8 @@ public override void AfterReturn(IBraceCompletionSession session, CancellationTo
...
                 if (document != null)
                 {
-                    // first add one line in between, and format braces 
-                    if (session.SubjectBuffer.GetOption(FeatureOnOffOptions.AutoFormattingOnCloseBrace))
-                    {
-                        document.InsertText(session.ClosingPoint.GetPosition(session.SubjectBuffer.CurrentSnapshot) - 1, Environment.NewLine, cancellationToken);
-                        FormatTrackingSpan(session, GetFormattingRules(document));
-                    }
+                    document.InsertText(session.ClosingPoint.GetPosition(session.SubjectBuffer.CurrentSnapshot) - 1, Environment.NewLine, cancellationToken);
+                    FormatTrackingSpan(session, false, GetFormattingRules(document));
```
Comment (balajikris) [inline src/EditorFeatures/CSharp/AutomaticCompletion/Sessions/CurlyBraceCompletionSession.cs]: Same comment as above. consider using named parameter

**Comment by balajikris** [comment]
:+1:

File: src/EditorFeatures/CSharp/AutomaticCompletion/Sessions/CurlyBraceCompletionSession.cs [hunk 3]
```diff
@@ -114,9 +110,9 @@ private IEnumerable<IFormattingRule> GetFormattingRules(Document document)
             return SpecializedCollections.SingletonEnumerable(BraceCompletionFormattingRule.Instance).Concat(Formatter.GetDefaultFormattingRules(document));
         }
 
-        private void FormatTrackingSpan(IBraceCompletionSession session, IEnumerable<IFormattingRule> rules = null)
+        private void FormatTrackingSpan(IBraceCompletionSession session, bool isAfterSessionStart, IEnumerable<IFormattingRule> rules = null)
```
Comment (Pilchie) [inline src/EditorFeatures/CSharp/AutomaticCompletion/Sessions/CurlyBraceCompletionSession.cs]: Is there a better name than this?  This encodes who the callers are, not how it changes the behavior of this method.

**Pull request merged**